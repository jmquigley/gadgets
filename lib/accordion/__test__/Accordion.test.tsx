'use strict';

import * as assert from 'assert';
import {shallow} from 'enzyme';
import * as React from 'react';
import {Accordion, getDefaultAccordionProps} from '../index';

test('Test retrieval of Accordion props object', () => {
	const props = getDefaultAccordionProps();

	assert(props);

	assert('children' in props);
	assert.equal(props.children, null);
});

test('Test the creation of a Accordion control container', () => {
	const li = <li>some list item</li>;
	const ctl = shallow(
		<Accordion id="1234" className="test-class">
			{li}
		</Accordion>
	);

	assert(ctl);
	expect(ctl).toMatchSnapshot();
});

// TODO: disabling the container
// TODO: make the container invisible
