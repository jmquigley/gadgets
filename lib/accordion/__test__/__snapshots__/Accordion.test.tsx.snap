// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test making the accordion control invisible 1`] = `
<ul
  className="ui-accordion accordion normal test-class invisible"
  style={Object {}}
>
  <li>
    some list item
  </li>
</ul>
`;

exports[`Test retrieval of Accordion props object 1`] = `
Object {
  "children": null,
  "className": "",
  "contentEditable": false,
  "direction": "right",
  "disabled": false,
  "focus": false,
  "height": "",
  "id": "",
  "location": "none",
  "noedit": false,
  "nohover": false,
  "noripple": false,
  "selected": false,
  "sizing": "normal",
  "style": Object {},
  "testing": false,
  "visible": true,
  "width": "",
}
`;

exports[`Test the creation of a Accordion control container 1`] = `
<ul
  className="ui-accordion accordion normal test-class"
  style={Object {}}
>
  <li>
    some list item
  </li>
</ul>
`;

exports[`Test the disabling of the Accordion control container 1`] = `
<ul
  className="ui-accordion accordion normal test-class disabled nohover"
  style={Object {}}
>
  <li>
    some list item
  </li>
</ul>
`;
